src/wasm/wasm-type.cpp

```diff
@@ -228,7 +228,7 @@ namespace {
 
 HeapTypeInfo* getHeapTypeInfo(HeapType ht) {
   assert(!ht.isBasic());
-  return (HeapTypeInfo*)ht.getID();
+  return (HeapTypeInfo*)(ht.with(Inexact).getID());
```

Comment:
Why is this correct?

Reply:
This clears the exact bit in the ID, recovering the value of the pointer to the underlying HeapTypeInfo.

Reply:
Then I'm surprised we don't need to clear out the other 2 reserved bits? In what way is bit 2 different than 0-1?

Reply:
Bits 0-1 are used in the `Type` representation and the `RecGroup` representation, but are always zero for `HeapType`.

Reply:
I see, thanks.

Perhaps we can add a helper for zeroing out the relevant bits?  That would be clearer than `with(Inexact)` and it would also be much easier to update in the future if we add more special bits.

Reply:
See the [rawID](https://github.com/WebAssembly/binaryen/pull/7396/commits/dcf0a1758c03525c6b4ad3ff6806abdaa621d27b) commit for this.